processing/flow_extractor_test.go

```diff
@@ -142,6 +142,7 @@ CheckLoop:
 				}
 				flows = append(flows, fe)
 				if len(flows) == len(expectedFlows) {
+					resultsLock.Unlock()
```

Comment:
There is an other break at line 141 where the unlocking does not happen when unmarshalling fails.

I believer you better factor this for-loop body out as a function, and use a defer in that function.

Reply:
> There is an other break at line 141 where the unlocking does not happen when unmarshalling fails.

That one is fine, as it does not `break` across multiple loop levels.

> I believe you better factor this for-loop body out as a function, and use a defer in that function.

Hmm, yes, That would then simply be a `return` which would trigger the unlock. I can try that, good idea.

Reply:
Done.